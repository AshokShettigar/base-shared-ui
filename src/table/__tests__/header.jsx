import { shallow } from 'enzyme';
import Column from 'platform-common-ui/table/column';
import Header from 'platform-common-ui/table/header';

describe('table/header', function() {
    const columns = [
        <Column id="1" key="1">Head 1</Column>,
        <Column id="2" key="2">Head 2</Column>
    ];

    it('should render the component', function() {
        const wrapper = shallow(<Header columns={columns} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component with scrollLeft', function() {
        const wrapper = shallow(<Header columns={columns} scrollLeft={100} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component with widths', function() {
        const widths = [80, 120];
        const wrapper = shallow(<Header columns={columns} widths={widths} />);
        expect(wrapper).toMatchSnapshot();
    });
});
