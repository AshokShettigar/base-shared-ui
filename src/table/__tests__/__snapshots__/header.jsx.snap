// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table/header should render the component 1`] = `
<table
  className="cn-table__header"
  data-component="Header"
  style={
    Object {
      "left": 0,
    }
  }
>
  <thead>
    <tr>
      <TableColumn
        id="1"
        isSortable={false}
        width="100px"
      >
        Head 1
      </TableColumn>
      <TableColumn
        id="2"
        isSortable={false}
        width="100px"
      >
        Head 2
      </TableColumn>
    </tr>
  </thead>
</table>
`;

exports[`table/header should render the component with scrollLeft 1`] = `
<table
  className="cn-table__header"
  data-component="Header"
  style={
    Object {
      "left": 100,
    }
  }
>
  <thead>
    <tr>
      <TableColumn
        id="1"
        isSortable={false}
        width="100px"
      >
        Head 1
      </TableColumn>
      <TableColumn
        id="2"
        isSortable={false}
        width="100px"
      >
        Head 2
      </TableColumn>
    </tr>
  </thead>
</table>
`;

exports[`table/header should render the component with widths 1`] = `
<table
  className="cn-table__header"
  data-component="Header"
  style={
    Object {
      "left": 0,
    }
  }
>
  <thead>
    <tr>
      <TableColumn
        id="1"
        isSortable={false}
        width="80px"
      >
        Head 1
      </TableColumn>
      <TableColumn
        id="2"
        isSortable={false}
        width="120px"
      >
        Head 2
      </TableColumn>
    </tr>
  </thead>
</table>
`;
